src/Microsoft.ML.TimeSeries/SrCnnEntireTransformer.cs

```diff
@@ -0,0 +1,903 @@
+                }
+                else if (_parent.OutputLength == 7)
+                {
+                    _slotNames = new VBuffer<ReadOnlyMemory<char>>(_parent.OutputLength, new[] { "Is Anomaly".AsMemory(), "Anomaly Score".AsMemory(), "Mag".AsMemory(),
+                        "Expected Value".AsMemory(), "Boundary Unit".AsMemory(), "Upper Boundary".AsMemory(), "Lower Boundary".AsMemory() });
+                }
+                else if (_parent.OutputLength == 4)
+                {
+                    _slotNames = new VBuffer<ReadOnlyMemory<char>>(_parent.OutputLength, new[] { "Is Anomaly".AsMemory(), "Raw Score".AsMemory(), "Mag".AsMemory(),
+                        "Expected Value".AsMemory() });
+                }
+            }
+
+            protected override DataViewSchema.DetachedColumn[] GetOutputColumnsCore()
+            {
+                var meta = new DataViewSchema.Annotations.Builder();
+                meta.AddSlotNames(_parent.OutputLength, GetSlotNames);
+                var info = new DataViewSchema.DetachedColumn[1];
+                info[0] = new DataViewSchema.DetachedColumn(_parent.OutputColumnName, new VectorDataViewType(NumberDataViewType.Double, _parent.OutputLength), meta.ToAnnotations());
+                return info;
+            }
+
+            public void GetSlotNames(ref VBuffer<ReadOnlyMemory<char>> dst) => _slotNames.CopyTo(ref dst, 0, _parent.OutputLength);
+
+            protected override Delegate MakeGetter(DataViewRow input, int iinfo, Func<int, bool> activeOutput, out Action disposer)
+            {
+                Contracts.AssertValue(input);
+                Contracts.Assert(0 <= iinfo && iinfo < _parent.ColumnPairs.Length);
+
+                var src = default(SrCnnTsPoint);
+                var getSrc = input.GetGetter<SrCnnTsPoint>(input.Schema[ColMapNewToOld[iinfo]]);
+
+                disposer = null;
+
+                ValueGetter<VBuffer<double>> del =
+                    (ref VBuffer<double> dst) =>
+                    {
+                        getSrc(ref src);
+                        if (src == null)
+                            return;
+                        var result = VBufferEditor.Create(ref dst, _parent.OutputLength);
+                        result.Values.Fill(Double.NaN);
+                        Double[] values = new Double[_parent.OutputLength];
+                        if (_parent.AnomalyModel.Inference(src, out values))
+                        {
+                            for (int i = 0; i < result.Values.Length; ++i)
+                            {
+                                result.Values[i] = values[i];
+                            }
+                        }
+                        dst = result.Commit();
+                    };
+
+                return del;
+            }
+        }
+
+        internal class SrCnnEntireModeler : ICanSaveModel
```

Comment:
>internal [](start = 8, length = 8)

This class can be private.

Also, the way I understand it, the only method in this class that is used externally is `Inference`, and that method only calls `TryGetValue` into a dictionary. I would consider changing this to a static class that has a static method `Train`, that returns the dictionary itself, and having the transformer use that dictionary directly (and be in charge of saving/loading it).

Reply:
> > internal [](start = 8, length = 8)
> 
> This class can be private.
> 
> Also, the way I understand it, the only method in this class that is used externally is `Inference`, and that method only calls `TryGetValue` into a dictionary. I would consider changing this to a static class that has a static method `Train`, that returns the dictionary itself, and having the transformer use that dictionary directly (and be in charge of saving/loading it).

Updated as you suggested.